crates/uv/src/commands/run.rs

```diff
@@ -1,29 +1,55 @@
+use crate::commands::reporters::{DownloadReporter, InstallReporter, ResolverReporter};
+use crate::commands::ExitStatus;
+use crate::commands::{elapsed, ChangeEvent, ChangeEventKind};
+use crate::printer::Printer;
+use anyhow::{Context, Result};
+use distribution_types::{IndexLocations, InstalledMetadata, LocalDist, Name, Resolution};
+use install_wheel_rs::linker::LinkMode;
+use itertools::Itertools;
+use owo_colors::OwoColorize;
+use pep508_rs::{MarkerEnvironment, PackageName, Requirement};
+use platform_tags::Tags;
+use pypi_types::Yanked;
 use std::ffi::OsString;
+use std::fmt::Write;
 use std::{env, iter};
-
-use anyhow::Result;
-use owo_colors::OwoColorize;
 use tempfile::{tempdir_in, TempDir};
+use tokio::process::Command;
 use tracing::debug;
-use uv_fs::Simplified;
-use uv_interpreter::PythonEnvironment;
 
-use crate::commands::ExitStatus;
-use tokio::process::Command;
 use uv_cache::Cache;
+use uv_client::{BaseClientBuilder, RegistryClient, RegistryClientBuilder};
+use uv_configuration::{
+    ConfigSettings, Constraints, NoBinary, NoBuild, Overrides, Reinstall, SetupPyStrategy,
+};
+use uv_dispatch::BuildDispatch;
+use uv_fs::Simplified;
+use uv_installer::{Downloader, Plan, Planner, SitePackages};
+use uv_interpreter::{Interpreter, PythonEnvironment};
+use uv_requirements::{
+    ExtrasSpecification, LookaheadResolver, NamedRequirementsResolver, RequirementsSource,
+    RequirementsSpecification, SourceTreeResolver,
+};
+use uv_resolver::{
+    Exclusions, FlatIndex, InMemoryIndex, Manifest, Options, OptionsBuilder, ResolutionGraph,
+    Resolver,
+};
+use uv_types::{BuildIsolation, EmptyInstalledPackages, HashStrategy, InFlight};
```

Comment:
Desperate to run IntelliJ's "Optimize imports" on this baby, to break into logical categories.

Reply:
lol I started fixing it then was like yuck merge conflicts will fix later